File: nim/difference-of-squares/difference_of_squares.nim
import math, sequtils

proc squareOfSum*(n: int): int =
  (1..n).toSeq.sum ^ 2

proc sumOfSquares*(n: int): int =
  (1..n).toSeq.mapIt(it ^ 2).sum

proc difference*(n: int): int =
  squareOfSum(n) - sumOfSquares(n)

File: prolog/difference-of-squares/difference_of_squares.pl
:- use_module(library(clpfd)).

square_of_sum(N, Result) :- Result #= (N * (N+1) // 2) ^ 2.
sum_of_squares(N, Result) :- Result #= (N * (N+1) * (2*N+1)) // 6.

difference(N, Result) :-
    square_of_sum(N, SquareOfSum),
    sum_of_squares(N, SumOfSquares),
    Result #= SquareOfSum - SumOfSquares.
